lib/rubocop/cop/rspec/scattered_setup.rb

```diff
@@ -23,25 +23,38 @@ module RSpec
       #   end
       #
       class ScatteredSetup < Cop
-        MSG = 'Do not define multiple hooks in the same example group.'
+        MSG = 'Do not define multiple `%<hook_name>s` hooks in the same '\
+              'example group (defined on lines %<lines>s).'
 
         def on_block(node)
           return unless example_group?(node)
 
-          analyzable_hooks(node).each do |repeated_hook|
-            add_offense(repeated_hook)
+          repeated_hooks(node).each do |hook_occurrences|
+            lines = hook_occurrences.map(&:first_line).join(', ')
+
+            hook_occurrences.each do |hook_occurrence|
+              add_offense(hook_occurrence,
+                          message: format_message(occurrence: hook_occurrence,
+                                                  lines: lines))
+            end
           end
         end
 
-        def analyzable_hooks(node)
-          RuboCop::RSpec::ExampleGroup.new(node)
+        def repeated_hooks(node)
+          hooks = RuboCop::RSpec::ExampleGroup.new(node)
             .hooks
             .select(&:knowable_scope?)
             .group_by { |hook| [hook.name, hook.scope, hook.metadata] }
             .values
             .reject(&:one?)
-            .flatten
-            .map(&:to_node)
+
+          hooks.map do |hook|
+            hook.map(&:to_node)
+          end
+        end
```

Comment:
This could have been

```diff
         def repeated_hooks(node)
-          hooks = RuboCop::RSpec::ExampleGroup.new(node)
+          RuboCop::RSpec::ExampleGroup.new(node)
             .hooks
             .select(&:knowable_scope?)
             .group_by { |hook| [hook.name, hook.scope, hook.metadata] }
             .values
             .reject(&:one?)
-
-          hooks.map do |hook|
-            hook.map(&:to_node)
-          end
+            .map do |hook|
+              hook.map(&:to_node)
+            end
         end
```

but I thought it was more readable as written. I don't have a strong feeling either way.